src/formats/notion-api/attachment-helpers.ts

```diff
@@ -0,0 +1,358 @@
+ */
+export function getCaptionFromBlock(block: any): string {
+	const blockType = block.type;
+	let captionArray: RichTextItemResponse[] = [];
+	
+	if (blockType === 'image' && block.image.caption) {
+		captionArray = block.image.caption;
+	}
+	else if (blockType === 'video' && block.video.caption) {
+		captionArray = block.video.caption;
+	}
+	else if (blockType === 'file' && block.file.caption) {
+		captionArray = block.file.caption;
+	}
+	else if (blockType === 'pdf' && block.pdf.caption) {
+		captionArray = block.pdf.caption;
+	}
+	else if (blockType === 'bookmark' && block.bookmark.caption) {
+		captionArray = block.bookmark.caption;
+	}
+	else if (blockType === 'link_preview' && block.link_preview.caption) {
+		captionArray = block.link_preview.caption;
+	}
+	else if (blockType === 'embed' && block.embed.caption) {
+		captionArray = block.embed.caption;
+	}
+	
+	// Convert rich text to plain text
+	return captionArray.map(t => t.plain_text).join('') || '';
+}
+
+/**
+ * Format attachment link according to vault settings
+ * @param result - Attachment download result
+ * @param vault - Obsidian vault
+ * @param caption - Optional caption/alt text
+ * @param isEmbed - Whether to use embed syntax (!) for images/videos/pdfs
+ * @returns Formatted markdown link
+ */
+export function formatAttachmentLink(
+	result: AttachmentResult,
+	vault: Vault,
+	caption: string = '',
+	isEmbed: boolean = false
+): string {
+	// If not local (still a URL), use standard markdown syntax
+	if (!result.isLocal) {
+		if (isEmbed) {
+			return `![${caption}](${result.path})`;
+		}
+		else {
+			return `[${caption || 'Link'}](${result.path})`;
+		}
+	}
+	
+	// Get user's link format preference
+	const useWikiLinks = vault.getConfig('useWikiLinks') ?? true;
+	
+	// Determine display text: prioritize caption, fallback to filename
+	const displayText = caption || result.filename || result.path;
+	
+	if (useWikiLinks) {
+		// Use Obsidian wiki link format
+		const embedPrefix = isEmbed ? '!' : '';
+		
+		// For wiki links, we need to include the file extension
+		// Obsidian requires the extension to properly link to non-markdown files
+		const pathWithExt = result.filename ? `${result.path}.${getFileExtension(result.filename)}` : result.path;
+		
+		if (caption) {
+			// If caption exists, use it as display text
+			return `${embedPrefix}[[${pathWithExt}|${caption}]]`;
+		}
+		else {
+			// If no caption, just use the file path with extension
+			return `${embedPrefix}[[${pathWithExt}]]`;
+		}
+	}
+	else {
+		// Use standard markdown format
+		// Need to add extension back for markdown links
+		const pathWithExt = result.filename ? `${result.path}.${getFileExtension(result.filename)}` : result.path;
+		
+		if (isEmbed) {
+			return `![${displayText}](${pathWithExt})`;
+		}
+		else {
+			return `[${displayText}](${pathWithExt})`;
+		}
+	}
+}
+
+/**
+ * Get file extension from filename
+ */
+function getFileExtension(filename: string): string {
```

Comment:
This can be replaced with `filesystem.splitext`.

Reply:
You are right. Done: https://github.com/obsidianmd/obsidian-importer/pull/444/commits/1248bb9b80b37653b5c61cb16529920dca4fad6a